osc/credentials/credentials.go

```diff
@@ -0,0 +1,113 @@
+package credentials
+
+import (
+	"os"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"encoding/json"
+	"io/ioutil"
+)
+
+var STD_PATH = filepath.Join(UserHomeDir(), ".osc", "config.json")
+
+type Credentials struct {
+	AccessKey string	`json:"access_key"`
+	SecretKey string	`json:"secret_key"`
+	RegionName string	`json:"region_name"`
+	Profile string
+}
+
+func NewCredentials(ak, sk, region, profile string) *Credentials {
+        return &Credentials{ AccessKey:ak, SecretKey:sk, RegionName:region, Profile:profile}
+}
+
+func (e *Credentials) CredentialsFromEnv() (Credentials, error) {
+
+	var err error
+        if os.Getenv("OSC_ACCESS_KEY") != "" {
+                e.AccessKey = os.Getenv("OSC_ACCESS_KEY")
+        } else {
+		err = fmt.Errorf("access key  not set")
+		return Credentials{}, err
+	}
+
+        if os.Getenv("OSC_SECRET_KEY") != "" {
+                e.SecretKey = os.Getenv("OSC_SECRET_KEY")
+        } else {
+		err := fmt.Errorf("secret key  not set")
+		return Credentials{}, err
+        }
+
+        if os.Getenv("OSC_REGION") != "" {
+                e.RegionName = os.Getenv("OSC_REGION")
+        } else {
+		err := fmt.Errorf("region  not set")
+		return Credentials{}, err
+        }
+
+return Credentials{
+        AccessKey: e.AccessKey,
+        SecretKey: e.SecretKey,
+        RegionName: e.RegionName,
+        }, err
+}
+
+func (p *Credentials) CredentialsFromFile(profile string) (Credentials, error) {
+        var Result map[string]Credentials
+        jsonFile, err := os.OpenFile(STD_PATH, os.O_RDONLY, 0755)
```

Comment:
Maybe it is better to use the os.Open() function because ith OpenFile, we specify the permissions however we do not want to create. os.Open() opens file in readonly.